v2/tools/generator/internal/codegen/pipeline/crossplane_add_for_provider.go

```diff
@@ -62,7 +62,7 @@ func nestSpecIntoForProvider(
 	// In the case where a spec type is reused across multiple resource definitions, we need to make sure
 	// to generate the same names for all of their nested properties, so base the nested type name off the
 	// spec type name
-	nestedTypeName := strings.Split(specName.Name(), "_")[0] + "Parameters"
+	nestedTypeName := strings.TrimSuffix(specName.Name(), astmodel.SpecSuffix) + "_Parameters"
```

Comment:
So, I'm not totally sold on doing this, because the `XParameters` is a Crossplane pattern and one that it feels like we don't really get a say in for generating resources for them. Did the original Swagger PR do this/is this needed for that PR? I so, we can live with it for now and undo it later when we figure out a better way to do naming. OTOH if this is just for consistency between the Crossplane types and the ASO types, I don't think we need that as badly as we need to match Crossplanes expected naming paradigm.

Reply:
Reversed, as discussed.